src/openms/source/ANALYSIS/MAPMATCHING/QTClusterFinder.cpp

```diff
@@ -615,6 +789,13 @@ void QTClusterFinder::createConsensusFeature_(ConsensusFeature& feature,
   {
     return feature_distance_(left->getFeature(), right->getFeature()).second;
   }
+
+  bool QTClusterFinder::distIsOutlier_(double dist, double rt)
+  {
+    if (bin_tolerances_.empty()) return false;
+    auto it = bin_tolerances_.lower_bound(rt);
+    return dist >= (--it)->second;
```

Comment:
- if bin_tolerances_ has exactly one element and rt is slightly smaller than this element. it will point on .begin(). --it is undefined. Can this happen (e.g. floating point) stuff?

Reply:
> Returns an iterator pointing to the first element in the range [first, last) that is not less than (i.e. greater or equal to) value, or last if no such element is found.

I interpreted it differently. I can check in an online compiler.

Reply:
https://godbolt.org/z/7jGdxz

Reply:
https://godbolt.org/z/arfGKx

Feel free to play around. Looks fine to me.

Reply:
I guess decrementing begin is implementation defined. I will add an if-case.